File: php/oneapi/response/ServiceException.php
<?php

class ServiceException {
	protected $messageId=null;
	protected $text=null;
	protected $variables=null;
			
	public function getMessageId() { return $this->messageId; }
	public function getText() { return $this->text; }
	public function getVariables() { return $this->variables; }
			
	public function setMessageId($messageId) { $this->messageId=(string) $messageId; }
	public function setText($text) { $this->text=(string) $text; }
	public function setVariables($variables) { $this->variables=(string) $variables; }
			
	public function __construct($jsondata) {
		if (!is_null($jsondata)) {
			if (property_exists($jsondata, 'messageId')) {
				$this->messageId = $jsondata->messageId;
			}
			if (property_exists($jsondata, 'text')) {
				$this->text = $jsondata->text;
			}
			if (property_exists($jsondata, 'variables')) {
				$this->variables = $jsondata->variables;
			}
		}
	}
	
	public function toString() {
		$buffer='messageId='.$this->messageId.', text='.$this->text.', variables=';
		if (!is_null($this->variables)) {
			if (is_array($this->variables)) {
				for ($i=0; $i<count($this->variables); $i++) {
					if ($i>0) {
						$buffer=$buffer.', ['.$i.'] = '.$this->variables[$i];
					} else {
						$buffer=$buffer.'['.$i.'] = '.$this->variables[$i];
					}
				}
			} else {
				$buffer=$buffer.$this->variables;
			}
		}
		return $buffer;
	}
		
}

?>

File: Ruby/response/ServiceException.rb

class ServiceException

	def initialize
		@messageId=nil
		@text=nil
		@variables=nil
	end

	def initializeJSON(jsondict)
		@messageId=nil
		if (jsondict!=nil) && (jsondict.has_key?'messageId') && (jsondict['messageId']!=nil)
			@messageId=jsondict['messageId']
		end
		@text=nil
		if (jsondict!=nil) && (jsondict.has_key?'text') && (jsondict['text']!=nil)
			@text=jsondict['text']
		end
		@variables=nil
		if (jsondict!=nil) && (jsondict.has_key?'variables') && (jsondict['variables']!=nil)
			@variables=jsondict['variables']
		end
	end

	def getMessageId
		@messageId
	end

	def setMessageId(messageId)
		@messageId=messageId
	end

	def getText
		@text
	end

	def setText(text)
		@text=text
	end

	def getVariables
		@variables
	end

	def setVariables(variables)
		@variables=variables
	end


end 
